x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteFilterAction.java

```diff
@@ -71,31 +81,31 @@ protected void doExecute(DeleteFilterAction.Request request, ActionListener<Ackn
                 }
             }
         }
-        if (!currentlyUsedBy.isEmpty()) {
-            throw ExceptionsHelper.conflictStatusException("Cannot delete filter, currently used by jobs: "
-                    + currentlyUsedBy);
-        }
+        return currentlyUsedBy;
+    }
 
-        DeleteRequest deleteRequest = new DeleteRequest(MlMetaIndex.INDEX_NAME, MlMetaIndex.TYPE, MlFilter.documentId(filterId));
+    private void deleteFilter(String filterId, ActionListener<AcknowledgedResponse> listener) {
+        DeleteRequest deleteRequest = new DeleteRequest(MlMetaIndex.INDEX_NAME, MlMetaIndex.TYPE,
+            MlFilter.documentId(filterId));
         BulkRequestBuilder bulkRequestBuilder = client.prepareBulk();
         bulkRequestBuilder.add(deleteRequest);
```

Comment:
Do you know why we use a bulk request containing a single request rather than using delete?

Reply:
I recall ES pushing towards always using bulk instead of isolated doc actions. I recall deprecations as well. Doesn't seem to be there now though, at least not for delete. In any case, I'd rather leave it as is to minimise churn.